cpp/src/neighbors/detail/cagra/add_nodes.cuh

```diff
@@ -136,9 +153,13 @@ void add_node_core(
         for (std::uint32_t i = 0; i < base_degree; i++) {
           std::uint32_t detourable_node_count = 0;
           const auto a_id                     = host_neighbor_indices(vec_i, i);
+          if (a_id >= idx.size()) {
+            detourable_node_count_list[i] = std::make_pair(a_id, base_degree + 1);
```

Comment:
It would be good to write a comment about this process, e.g., in what case this conditional branch becomes true and why the detourable node count is set to base_degree + 1.

Reply:
Thanks, I've added the comments.